### DISCUSSION TITLE
Force regenerate slug/permalink

### DISCUSSION THREAD

**Discussion by AmritasyaPutra** [idno/idno#3071] [Q&A] [answered]
On the home page https://k.ekvastra.in/, the first article is linked as https://k.ekvastra.in/2017/09/09/steps-to-attain-yoga---yama. I have edited the title and want the slug/permalink to be regenerated, how do I do that? I am doing this because I migrated from wordpress and I need the earlier (wordpress) links to remain valid after redirection and for that I had to edit some titles in withknown but I am unable to force it to regenerate a slug based on the edited new title.

**Comment by AmritasyaPutra** [comment]
I already tried editing slug under entities_metadata table, that moves the article to new url hence the old google search results redirect properly but the pages on the site break because they continue to have the old slug/permalink. I have undone my manual changes to the table. Where else is slug/permalink stored - how do I trigger a universal regenerate of slug/permalink based on the edited (new) title?

**Comment by lindner** [comment, marked as answer]
The slugs are also stored inside the json in various tables IIRC.  It's a
bit painful to manually patch that all together...

**Comment by AmritasyaPutra** [comment]
I agree, I figured that out and I resorted to url redirect in cloudflare to redirect old links to the new one. Effectively I have achieved what I wanted through cloudflare bulk redirection. I am good here. Another way I thought of but could not try was to export my site and then import it again. That should have re-generated the slug based on the new titles during import. But I was not able to export my site and I could not test this idea. I was getting the following one-line error in php on export:

`[26-Apr-2022 07:13:45 UTC] Known (k.ekvastra.in): critical - Cannot instantiate interface Idno\Core\Template [/home/ekvastra/k.ekvastra.in/Idno/Core/Migration.php:590]`

I am on Version: 1.2.2+2020061101 and the above error points to the following code on the tip:
https://github.com/idno/known/blob/7af2cad1b0e5457dd8090a6e603562aef06d219e/Idno/Core/Migration.php#L621

Instead of DefaultTemplate I have Template there. I was able to export after I changed it to DefaultTemplate but I didn't bother to import and test as I have already put cloudflare bulk redirect in place.
